rock-paper-scissors/scripts/rock-paper-scissors.js

```diff
@@ -0,0 +1,158 @@
+const gameChoices = ["rock", "paper", "scissors"];
+let playerSelection = "";
+let computerSelection = "";
+
+let playerWins = 0;
+let computerWins = 0;
+let nobodyWins = 0;
+let roundCount = 0;
+
+//initiates the new round by calling startNewRound. startNewRound calls for
+// the player and computer functions to make a selection.
+
+startNewRound();
+function startNewRound() {
+  playerPlay();
+  computerPlay();
+  playGame(playerSelection, computerSelection);
+}
```

Comment:
So in an effort to reduce the number of global variables, what you were missing the last time we spoke was l15 and l16. They need to be:

```
  const playerSelection = playerPlay();
  const computerSelection = computerPlay();
```

With l2 and l3 gone. And the `playerPlay()` functions need to be:
```
function playerPlay() {
    return prompt("Please enter Rock, Paper, or scissors").toLowerCase();
}
```

The flow here is we store the result of the `playerPlay()` function into the new local variable `playerSelection` and pass that along.

Reply:
Some confusion here. With this it seems that because the variables are constant, I cannot reassign them on new rounds. So whatever my input is on the first round, it remains during following rounds. As well, I don't know why; but, it throws off the counter, some rounds go without triggering a win?

I have played with this one three different times now and I can't seem to get it to work right. Some rounds will play just fine but not all rounds play correctly.

Reply:
Declaring `const`s inside the function make them local scoped. So any new call of `startNewRound()` will reinitialise all of the variables, and require prompts. The code should look like this:

```
function playerPlay() {
    return prompt("Please enter Rock, Paper, or scissors").toLowerCase();
}

function startNewRound() {
  const playerSelection = playerPlay();
  const computerSelection = computerPlay();
  playGame(playerSelection, computerSelection);
}
```

Does that not work?